src/types/list.md

```diff
@@ -1 +1,66 @@
 # List
+
+Lists are the SML type of ordered collections of objects. Notably, you can create lists of any kind of object, so `int list`, `string list`, and `bool list` are all valid types. `list` on its own is a type constructor (meaning that it makes new types out of old types), so it is not a valid type by itself, however. Lists are not, however, the same as arrays - they do not have constant-time access to any given index of the list. Instead, you only have access to the elements located at the very beginning of the list, the _head_. Additionally, lists are more _restricted_ than data structures in some other languages - a given list has a fixed type for its elements. All of the elements in a list must be of the same type. For the purposes of this document, we will discuss only int lists.
```

Comment:
Is there an advantage to sticking to int lists? I think it might be a good idea to familiarize them with polymorphic structures early

Reply:
I asked Jacob, and he said that he'd recommend not introducing polymorphic lists yet. I was going for a page for lists that would be understandable to a student in the course right now, hence only discussion of int lists.

Reply:
Well, we can discuss lists without needing to talk about `'a list` as a type. Instead of talking about polymorphism, we could just mention that int list, bool list, etc, are all types. I think it's important to show them early examples of multiple different list types (with bools, nested lists, etc).

Reply:
I've made a change that should have some mention of arbitrary typed lists now.